src/block_store.rs

```diff
@@ -26,14 +32,21 @@ impl BlockStore {
             Self::fetch_latest(rpc_client, CommitmentConfig::finalized()).await?;
 
         Ok(Self {
-            latest_confirmed_blockinfo: Arc::new(RwLock::new(confirmed_block.clone())),
-            latest_finalized_blockinfo: Arc::new(RwLock::new(finalized_block.clone())),
+            latest_confirmed_block: Arc::new(RwLock::new((
+                confirmed_blockhash.clone(),
+                confirmed_block,
+            ))),
+            latest_finalized_block: Arc::new(RwLock::new((
+                finalized_blockhash.clone(),
+                finalized_block,
+            ))),
             blocks: Arc::new({
                 let map = DashMap::new();
                 map.insert(confirmed_blockhash, confirmed_block);
                 map.insert(finalized_blockhash, finalized_block);
                 map
             }),
+            last_add_block_metric: Arc::new(RwLock::new(Instant::now())),
```

Comment:
remove the add_block_metric logic from this PR. As you have introduced in another PR.

Reply:
Yeah this can be resolved during merging